library/test/src/test/java/com/bumptech/glide/load/engine/executor/GlideExecutorTest.java

```diff
@@ -49,6 +55,46 @@ public void onRun(int priority) {
     assertThat(resultPriorities).containsExactly(5, 1, 2, 3, 4).inOrder();
   }
 
+  @Test
+  public void newTestExecutor_teardown_isShutdown() {
+    ExecutorService executor = Executors.newSingleThreadExecutor();
+    try {
+      GlideExecutor glideExecutor = MockGlideExecutor.newTestExecutor(executor);
+      Glide.init(
+          ApplicationProvider.getApplicationContext(),
+          new GlideBuilder()
+              .setAnimationExecutor(glideExecutor)
+              .setDiskCacheExecutor(glideExecutor)
+              .setSourceExecutor(glideExecutor));
+
+      Glide.tearDown();
+
+      assertThat(executor.isShutdown()).isTrue();
+    } finally {
+      executor.shutdown();
+    }
+  }
+
+  @Test
+  public void wrapExecutor_teardown_isNotShutdown() {
+    ExecutorService executor = Executors.newSingleThreadExecutor();
+    try {
+      GlideExecutor glideExecutor = MockGlideExecutor.wrapExecutor(executor);
+      Glide.init(
+          ApplicationProvider.getApplicationContext(),
+          new GlideBuilder()
+              .setAnimationExecutor(glideExecutor)
+              .setDiskCacheExecutor(glideExecutor)
+              .setSourceExecutor(glideExecutor));
+
+      Glide.tearDown();
+
+      assertThat(executor.isShutdown()).isFalse();
+    } finally {
+      executor.shutdown();
```

Comment:
Slightly better to use https://github.com/bumptech/glide/blob/26ecf64550a07d1b8f958a8e3ef192028b4892ca/library/src/main/java/com/bumptech/glide/util/Executors.java#L41 because it will wait if any tasks are scheduled.

Really there shouldn't be anything on the executor, but conceivably there could be some initialization task at some point that leaks across tests via the singleton.